add swift-format and setup gh lint action

JaapWijnen opened the pull request:
Adds the apple/swift-format plugin and sets up a lint pass through GitHub actions

This currently runs on a macOS runner since it's easier to install swift-format through brew. Can easily be moved to linux once swift-format plugins work again with the --configuration argument
https://github.com/apple/swift-format/issues/668

JaapWijnen committed:
add swift-format and setup gh lint action

JaapWijnen committed:
format

BradLarson commented on Package.swift:
```diff
@@ -7,13 +7,20 @@ let package = Package(
+        .executable(name: "CustomDerivatives", targets: ["CustomDerivatives"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/apple/swift-format.git", .upToNextMajor(from: "509.0.0"))
```
Why do we need `swift-format` as a dependency? Do we get easy access to a formatting plugin as a result?

JaapWijnen replied:
Oh that's a good call. I started out wanting to run it by using the plugin but in the end I'm using swift-format installed through brew since there's a bug in either command plugins or the swift-format plugin where I can't pass it a configuration file (https://github.com/apple/swift-format/issues/668) 
We can move to use the plugin and then also run the CI on linux once that is solved. But for now running on macOS and installing through brew was easier. I'll remove the dependency for now.

JaapWijnen committed:
remove swift-format dependency

Since we're not using the swift-format package plugin for formatting for now (due to a bug in passing a configuration file) we can remove the dependency again.

JaapWijnen committed:
fix readme links
